### ISSUE TITLE
Consider using a different default keybinding for `lsp-inline-completion-accept` in terminal or document it and recommend a different keybinding

### ISSUE THREAD

**Issue by xuhdev** [emacs-lsp/lsp-mode#4657]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

Terminals translate `Ctrl-Enter` to `Enter`. Currently, `lsp-inline-completion-accept` by default is bound to `Ctrl-Enter`. This is confusing to a user unfamiliar with terminal key pitfalls: The user follows the instructions and hits `Ctrl-Enter`, but actually Emacs receives `Enter`, which `lsp-inline-completion-mode` translates to "Cancel".

A more detailed explanation on keys in terminal emulators: https://unix.stackexchange.com/a/161725/38242

### Steps to reproduce

Launch a terminal, try to accept a completion generated by `lsp-inline-completion-mode` with `Ctrl-Enter`.

### Expected behavior

Either inform that they should use a different keybinding, or use a different default keybinding in terminal.

### Which Language Server did you use?

N/A

### OS

Linux

**Comment by BaoshanPang** [comment]
Agree. I finally make it work by changing the 'C-return' to 'M-RET'. Thanks for the hint.

**Comment by williewillus** [comment]
Just ran into this myself without knowing about the pitfalls of terminal keys. I'd suggest providing `TAB` or as mentioned above M-RET for terminal users.
